### PR TITLE
ISPN-16239: Add documentation about using ISPN in WildFly applications

### PR DESCRIPTION
**Pull request by theashiot**
https://issues.redhat.com/browse/ISPN-16239

### CODE REVIEW

**Review by wfink — CHANGES_REQUESTED** [review]
It is not the comlete picture.
There are 2 possibilities
1) use the EAP bits (need to have jboss-structure.xml
2) include the RHDG bits into the application, This is what you describe here

**Comment by wfink** [comment]
EAP8 is different, so I need to play with it and upgrade my example to EAP8 without modules

**Comment by tristantarrant** [comment]
@wfink this is waiting on you

File: documentation/src/main/asciidoc/topics/dependencies_maven/eap_dependencies.xml [hunk 1]
```diff
@@ -2,26 +2,12 @@
   <dependency>
     <groupId>org.infinispan</groupId>
     <artifactId>infinispan-core</artifactId>
-    <scope>provided</scope>
+    <scope>compile</scope>
```
Comment (wfink) [inline documentation/src/main/asciidoc/topics/dependencies_maven/eap_dependencies.xml:5]: provided is correct here as you on't want to include the bits if using WF

File: documentation/src/main/asciidoc/topics/dependencies_maven/eap_dependencies.xml [hunk 2]
```diff
@@ -2,26 +2,12 @@
...
   </dependency>
   <dependency>
     <groupId>org.infinispan</groupId>
     <artifactId>infinispan-cachestore-jdbc</artifactId>
-    <scope>provided</scope>
```
Comment (wfink) [inline documentation/src/main/asciidoc/topics/dependencies_maven/eap_dependencies.xml:10]: keep provided

File: documentation/src/main/asciidoc/topics/proc_adding_dependencies_wildfly_modules.adoc [hunk 3]
```diff
@@ -0,0 +1,14 @@
+[id='configure_ispn_modules']
+= Configuring Applications to Use {brandname} Modules
+
+To use {brandname} in  {wildflybrandname} applications, add {brandname} dependency in the application's pom.xml file. 
+
+.Procedure
+
+* In your project `pom.xml` file, mark the required {brandname} dependencies as _compile_.
```
Comment (wfink) [inline documentation/src/main/asciidoc/topics/proc_adding_dependencies_wildfly_modules.adoc:8]: wrong ...

File: documentation/src/main/asciidoc/stories/assembly_wildfly_modules.adoc [hunk 4]
```diff
@@ -0,0 +1,11 @@
+[id='ispn_modules']
+:context: server
+= Using {brandname} in {wildflybrandname} applications
+
+{wildflybrandname} includes {brandname} modules that you can use in {wildflybrandname} applications.
```
Comment (wfink) [inline documentation/src/main/asciidoc/stories/assembly_wildfly_modules.adoc:5]: not sure whether we should mention the WF version which includes core only and from which version the full API is available (like query counter etc)
Or we simple rely on the assumption that it is the latest

**Review by wfink — CHANGES_REQUESTED** [review]
@theashiot I've created the example [1] a while ago and assumed you will integrate that here.

[1] https://github.com/wfink/jboss-eap-playground/tree/master/EAP8cache

File: documentation/src/main/asciidoc/topics/dependencies_maven/eap_dependencies.xml (hunk 2 above)
Comment (theashiot) [reply documentation/src/main/asciidoc/topics/dependencies_maven/eap_dependencies.xml:10]: fixed

File: documentation/src/main/asciidoc/topics/proc_adding_dependencies_wildfly_modules.adoc (hunk 3 above)
Comment (theashiot) [reply documentation/src/main/asciidoc/topics/proc_adding_dependencies_wildfly_modules.adoc]: fixed

File: documentation/src/main/asciidoc/stories/assembly_wildfly_modules.adoc (hunk 4 above)
Comment (theashiot) [reply documentation/src/main/asciidoc/stories/assembly_wildfly_modules.adoc]: I think it'd be good to mention the versions. Can you please help me with the info?

**Comment by theashiot** [comment]
Thanks @wfink for the example and the review, and apologies for the delay. Do you think the described steps and information is adequate?

best,
ashwin

File: .asciidoctorconfig.adoc [hunk 5]
```diff
@@ -0,0 +1,12 @@
+// +++++++++++++++++++++++++++++++++++++++++++++++++++++++
```
Comment (tristantarrant) [inline .asciidoctorconfig.adoc:1]: This file should be removed

**Comment by ryanemerson** [comment]
Thanks @theashiot

**Pull request merged**